### DISCUSSION TITLE
Best way to manage multiple modules?

### DISCUSSION THREAD

**Discussion by t-huyeng** [cyberman54/ESP32-Paxcounter#927] [Q&A] [answered]
I am wondering how other people dealing with my current problem: 

I have multiple different ESP-Boards which I frequently want to update or I want to try out different stuff on some of the beards. If switch between boards I currently have to manually change the LoRaWAN Keys to the correct ones for each board. 

I do not really want to have a separate branch for each board or even the code copied for each board. Would it be possible to work with environment variables and different set ups in the platformio.ini or has somebody a better way of doing this?

Due to other circumstance I do not think that OTA is a choice for me. 

Thanks

**Comment by cyberman54** [comment, marked as answer]
Easiest way is to use OTAA and same APPKEY and APPEIU for all boards. Using an all-zero DEVEIU (see example in loraconf.h), then each board will generate it's own unique DEVEIU while joining, based on it's internal unique ESP32 MAC. 

I would in general not recommend to use ABP, because it has some disadvantages.

**Reply by t-huyeng** [reply]
That is a really good point. Thanks! 

I will try to use the same APPKEY and APPEIU and only use the generated DEVEUI, will report back tomorrow!

**Reply by t-huyeng** [reply]
Looks like it is working for me like that, even without OTA, much better to just leave the DEVEUI empty. It looks like the ESP is using the "old" ID, but this is also fine with me.

**Reply by cyberman54** [reply]
You may want to use the "new" format, i did not change this yet, because would be a hard breaking change for all existing installations:

https://github.com/cyberman54/ESP32-Paxcounter/blob/master/src/lorawan.cpp#L97
